File: MicroFrontend/appMicro/ios/App/App/plugins/StoreRating/StoreRatingPlugin.m
//
//  StoreRatingPlugin.m
//  App
//
//  Created by Jimmy Alfonso Cardenas Orjuela on 3/11/21.
//

#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(StoreRatingPlugin, "StoreRating",
           CAP_PLUGIN_METHOD(requestReview, CAPPluginReturnPromise);
)

File: MicroFrontend/appMicro/ios/App/App/plugins/StoreRating/StoreRatingPlugin.swift
//
//  StoreRatingPlugin.swift
//  App
//
//  Created by Jimmy Alfonso Cardenas Orjuela on 3/11/21.
//

import Foundation
import Capacitor
import StoreKit

@objc(StoreRatingPlugin)
public class StoreRatingPlugin: CAPPlugin, CAPBridgedPlugin {
    
    public let identifier = "StoreRatingPlugin"
    public let jsName = "StoreRating"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "configureStoreRating", returnType: CAPPluginReturnPromise)
    ]
    
    @objc func requestReview(_ call: CAPPluginCall) {
        SKStoreReviewController.requestReview()
        call.resolve()
    }
}

